yarn-project/foundation/src/fifo/memory_fifo.ts

```diff
@@ -57,15 +57,18 @@ export class MemoryFifo<T> {
   /**
    * Put an item onto back of the queue.
    * @param item - The item to enqueue.
+   * @returns A boolean indicating whether the item was successfully added to the queue.
    */
-  public put(item: T) {
+  public put(item: T): boolean {
```

Comment:
minor unrelated cleanup so we don't need to keep running flags to avoid inserting to a flushing queue.